tests/test_waiting.py

```diff
@@ -104,47 +104,58 @@ def test_wait_epoll_bad(pgconn):
     assert res.status == ExecStatus.TUPLES_OK
 
 
+@pytest.fixture
+def wait_async(anyio_backend_name):
+    return {
+        "asyncio": waiting.wait_asyncio,
+        "trio": waiting.wait_anyio,
+    }[anyio_backend_name]
+
+
+@pytest.fixture
+def wait_conn_async(anyio_backend_name):
+    return {
+        "asyncio": waiting.wait_conn_asyncio,
+        "trio": waiting.wait_conn_anyio,
+    }[anyio_backend_name]
+
+
 @pytest.mark.parametrize("timeout", timeouts)
-@pytest.mark.asyncio
-async def test_wait_conn_async(dsn, timeout):
+async def test_wait_conn_async(dsn, timeout, wait_conn_async):
     gen = generators.connect(dsn)
-    conn = await waiting.wait_conn_async(gen, **timeout)
+    conn = await wait_conn_async(gen, **timeout)
     assert conn.status == ConnStatus.OK
 
 
-@pytest.mark.asyncio
-async def test_wait_conn_async_bad(dsn):
+async def test_wait_conn_async_bad(dsn, wait_conn_async):
     gen = generators.connect("dbname=nosuchdb")
     with pytest.raises(psycopg.OperationalError):
-        await waiting.wait_conn_async(gen)
+        await wait_conn_async(gen)
 
 
-@pytest.mark.asyncio
-async def test_wait_async(pgconn):
+async def test_wait_async(pgconn, wait_async):
     pgconn.send_query(b"select 1")
     gen = generators.execute(pgconn)
-    (res,) = await waiting.wait_async(gen, pgconn.socket)
+    (res,) = await wait_async(gen, pgconn.socket)
     assert res.status == ExecStatus.TUPLES_OK
 
 
-@pytest.mark.asyncio
 @pytest.mark.parametrize("wait, ready", zip(waiting.Wait, waiting.Ready))
 @skip_if_not_linux
-async def test_wait_ready_async(wait, ready):
+async def test_wait_ready_async(wait, ready, wait_async):
```

Comment:
Is the `wait_async` fixture only used in this couple of tests? Maybe we could just test the `wait_asyncio()/wait_anyio()` functions in separate tests instead.

Reply:
We'd need to duplicate three test functions for `wait_async` and two more for `wait_conn_async`. That's beyond my (subjective) limit for code repetition :)